services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/RouteWithSubnetMapperServiceImpl.java

```diff
@@ -0,0 +1,153 @@
+
+import com.futurewei.alcor.common.db.CacheException;
+import com.futurewei.alcor.common.exception.DatabasePersistenceException;
+import com.futurewei.alcor.common.exception.ResourceNotFoundException;
+import com.futurewei.alcor.common.exception.ResourcePersistenceException;
+import com.futurewei.alcor.route.dao.RouteRepository;
+import com.futurewei.alcor.route.dao.RouteWithSubnetMapperRepository;
+import com.futurewei.alcor.route.service.RouteWithSubnetMapperService;
+import com.futurewei.alcor.web.entity.route.RouteEntity;
+import com.futurewei.alcor.web.entity.route.SubnetToRouteMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class RouteWithSubnetMapperServiceImpl implements RouteWithSubnetMapperService {
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Autowired
+    RouteWithSubnetMapperRepository routeWithSubnetMapperRepository;
+
+    @Autowired
+    RouteRepository routeRepository;
+
+    @Override
+    public SubnetToRouteMapper getBySubnetId(String subnetId) throws ResourceNotFoundException, ResourcePersistenceException {
+        try {
+            return this.routeWithSubnetMapperRepository.findItem(subnetId);
+        }catch (Exception e) {
+            return null;
+        }
+    }
+
+    @Override
+    public List<RouteEntity> getRuleBySubnetId(String subnetId) throws ResourceNotFoundException, ResourcePersistenceException {
+        List<RouteEntity> routes = new ArrayList<>();
+
+        try {
+            SubnetToRouteMapper subnetToRouteMapper =  this.routeWithSubnetMapperRepository.findItem(subnetId);
+            if (subnetToRouteMapper == null) {
+                return null;
+            }
+
+            List<String> routeIds = subnetToRouteMapper.getRouteIds();
+            for (String routeId : routeIds) {
+                RouteEntity route = this.routeRepository.findItem(routeId);
+                if (route != null) {
+                    routes.add(route);
+                }
+            }
+
+            return routes;
+
+        }catch (Exception e) {
+            return null;
+        }
+    }
+
+    @Override
+    public Map getAllMappers() throws CacheException {
+        return this.routeWithSubnetMapperRepository.findAllItems();
+    }
+
+    @Override
+    public void addMapper(SubnetToRouteMapper subnetToRouteMapper) throws DatabasePersistenceException {
+        try {
+            this.routeWithSubnetMapperRepository.addItem(subnetToRouteMapper);
+        } catch (Exception e) {
+            throw new DatabasePersistenceException(e.getMessage());
+        }
+    }
+
+    @Override
+    public void addMapperByRouteEntity(String subnetId, RouteEntity routeEntity) throws DatabasePersistenceException {
+        try {
```

Comment:
In Route Manager ServiceImpl and repo classes, I don't find usage of a transaction. 

Please find the reference implementation: https://github.com/futurewei-cloud/alcor/blob/7c6f138df9199e4437f0958a3b18287bcb3df09d/services/port_manager/src/main/java/com/futurewei/alcor/portmanager/repo/PortRepository.java#L66

@kevin-zhonghao

Reply:
Fix it ~

Reply:
Sure

Reply:
Create a ticket for tracking https://github.com/futurewei-cloud/alcor/issues/274